dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumResourceScopeTest.kt

```diff
@@ -177,6 +178,53 @@ internal class RumResourceScopeTest {
         assertThat(result).isEqualTo(null)
     }
 
+    @Test
+    fun `𝕄 send Resource with initial context 𝕎 handleEvent(StopResource)`(
+        @Forgery context: RumContext,
+        @Forgery kind: RumResourceKind,
+        @LongForgery(200, 600) statusCode: Long,
+        @LongForgery(0, 1024) size: Long,
+        forge: Forge
+    ) {
+        // Given
+        val attributes = forge.exhaustiveAttributes()
+        val expectedAttributes = mutableMapOf<String, Any?>()
+        expectedAttributes.putAll(fakeAttributes)
+        expectedAttributes.putAll(attributes)
+        whenever(mockParentScope.getRumContext()) doReturn context
```

Comment:
maybe we should add here : `whenever(mockParentScope.getRumContext()) doReturn context doReturn null`

Reply:
I'm not sure I get what you're trying to do with that ?

Reply:
Also we can't return null as `getRumContext()` return  type is non nullable

Reply:
I am trying to test that getRumContext is being called only once as now you are persisting the initialised value

Reply:
Maybe instead of null you  could return a Context with different properties the second time

Reply:
Well I don't really care if it's called more than once, the goal of the test is to make sure that in the written event, the initial context is used.

Reply:
Actually you had this there : `whenever(mockParentScope.getRumContext()) doReturn context` it is what I intended also but to put it in the `set up` method to make it more obvious. Sorry for the confusion I only saw this after I checked out your PR.